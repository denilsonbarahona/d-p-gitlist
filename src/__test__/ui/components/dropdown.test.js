import React from 'react';
import {mount} from 'enzyme'
import {create} from 'react-test-renderer'
import Dropdown from '../../../shared/components/dropdown'

describe('dropdown',()=>{
    const dropdown = mount(<Dropdown items={['option1', 'option2']}/>)

    test('render dropdown',()=>{
        expect(dropdown.length).toEqual(1)
    })

    test('render dropdown panel',()=>{
        const panel=dropdown.find('.dropdown-panel');
        expect(panel.length).toEqual(1)
    })

    test('simulate event click on item',()=>{
        const onEventClick = jest.fn()
        const ItemClick = mount(<Dropdown items={['option1', 'option2']} onSelected={onEventClick}/>)
        ItemClick.find('.dropdown-item').first().simulate('click')
        expect(onEventClick).toHaveBeenCalledTimes(1)
    })

    test('create snapshot',()=>{
        const dropdownSnapshot = create(<Dropdown items={['option1', 'option2']}/>)
        expect(dropdownSnapshot.toJSON()).toMatchSnapshot()
    })
})
