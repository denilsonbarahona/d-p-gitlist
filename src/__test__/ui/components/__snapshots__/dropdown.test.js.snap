// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown create snapshot 1`] = `
<div
  aria-label="select an option"
  className="dropdown"
  role="listbox"
>
  <div
    className="dropdown-title"
    onClick={[Function]}
    onKeyUp={[Function]}
    tabIndex="0"
  >
    <span />
    <i
      aria-hidden="true"
      className="icon-circle-down"
    />
  </div>
  <div
    className="dropdown-panel dropdown-panel--isHidden"
  >
    <div
      className="dropdown-header"
    >
      <p>
        Select 
      </p>
      <i
        aria-label="click to close to panel"
        className="icon-cross"
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      />
    </div>
    <div>
      <div
        className="dropdown-item dropdown-item--Selected"
        id="option1"
        onClick={[Function]}
        onKeyUp={[Function]}
        role="option"
        tabIndex="0"
      >
        <i
          className="icon-checkmark"
        />
        option1
      </div>
      <div
        className="dropdown-item dropdown-item--UnSelected"
        id="option2"
        onClick={[Function]}
        onKeyUp={[Function]}
        role="option"
        tabIndex="0"
      >
        <i
          className="icon-checkmark"
        />
        option2
      </div>
    </div>
  </div>
</div>
`;
